paramnb/__init__.py

```diff
@@ -263,7 +263,12 @@ def widgets(self):
         """Return name,widget boxes for all parameters (i.e., a property sheet)"""
 
         params = self.parameterized.params().items()
-        ordered_params = list(OrderedDict(sorted(params, key=lambda x: x[1].precedence if x[1].precedence else -100)).keys())
+        sorted_precedence = sorted(params().items(),
+                        key=lambda x: x[1].precedence if x[1].precedence else -float('inf'))
+        groups = itertools.groupby(sorted_precedence, key=lambda p: p[1].precedence
+                                   if p[1].precedence else -float('inf'))
```

Comment:
This should also be omitting parameters with a negative precedence, e.g. by slicing the list after it's sorted.  

Once negative precedences are omitted, then there's no particular reason to use float('inf') for precedence of None.  Maybe None should actually have a precedence of something intermediate, like 10, so that with the usual 0 to 1.0 precedence values, anything with no precedence goes after those with declared precedence, while still allowing people to force something to appear after all those with no declared precedence (by setting a precedence > 10)?  10 seems arbitrary, but there wouldn't seem to be any particular boundary that would make more sense than that...